### PR TITLE
adding payee info into Payment schema

### PR DESCRIPTION
**Pull request by olegromashin**

### CODE REVIEW

**Comment by vladwix** [comment]
Tagless sum types are actually unsafe, due to potential field name clashes. I'd recommend _object with single field_ encoding which is probably the best and is easy to understand. It looks something like this:
```yaml
       payee:
          type: object
          oneOf:
            - properties:
                person:
                  $ref: '#/components/schemas/Person'
            - description: Virtual private account
              properties:
                vpa:
                  type: string
            - description: Bank account
              properties: 
                bank_account:
                  type: object
                  properties:
                    ifsc_code:
                      type: string
                    account_number:
                      type: string
                    account_holder_name:
                      type: string
```

**Comment by olegromashin** [comment]
Agree, applied your suggestion.

**Comment by beckn-dev** [comment]
This will not be a core spec standard. Virtual payment addresses are part of an India specific payment interface. The payment terms of a BPP must contain the following attributes

1. What is the total amount that needs to be paid?
2. What amount has been paid?
3. What is the balance amount that is pending?
4. Breakup of the payment (bunch of key value pairs)
5. When should the payment happen - ON-ORDER, PRE-FULFILLMENT, ON-FULFILLMENT or  POST-FULFILLMENT?
6. What are the methods supported basis Point no 4
7. What is the last date of payment in case of POST-FULFILLMENT

The payment params required for each of the above method will be decided at a network level rather than at a spec level. So in the core spec, the params will be just a generic object. The supported keys of the params will be standardized when the network is being designed. 

Any late payment policies, refund policies etc will be mutually agreed upon in the Network Participation Agreement

In conclusion, the core payment schema must contain the above terms without specifically going into terms specific to a network or a region.

**Comment by olegromashin** [comment]
Decided to add fields currency as required and transaction_status as optional into params and remove payee object:
currency - $ref: '#/components/schemas/MonetaryValue/properties/currency'
transaction_status - string
